proc_dash/utility.py

```diff
@@ -44,28 +54,56 @@ def get_required_bagel_columns() -> list:
 
 # TODO: When possible values per column have been finalized (waiting on mr_proc),
 # validate that each column only has acceptable values
-def get_missing_required_columns(bagel: pd.DataFrame) -> set:
+def get_missing_required_columns(bagel: pd.DataFrame, schema_file: str) -> set:
     """Identifies any missing required columns in bagel schema."""
-    missing_req_columns = set(get_required_bagel_columns()).difference(
-        bagel.columns
-    )
+    missing_req_columns = set(
+        get_required_bagel_columns(schema_file)
+    ).difference(bagel.columns)
 
     return missing_req_columns
 
 
-def extract_pipelines(bagel: pd.DataFrame) -> dict:
+def get_event_id_columns(
+    bagel: pd.DataFrame, schema: str
+):  # TODO: Add type hint
+    """Returns names of columns which identify a unique assessment or processing pipeline."""
+    if schema == "imaging":
+        return ["pipeline_name", "pipeline_version"]
+    elif schema == "phenotypic":
+        return (
+            ["assessment_name", "assessment_version"]
+            if "assessment_version" in bagel.columns
+            else "assessment_name"
+        )
+
+
+def extract_pipelines(bagel: pd.DataFrame, schema: str) -> dict:
     """Get data for each unique pipeline in the aggregate input as an individual labelled dataframe."""
     pipelines_dict = {}
 
-    pipelines = bagel.groupby(["pipeline_name", "pipeline_version"])
-    for (name, version), pipeline in pipelines:
-        label = f"{name}-{version}"
-        # per pipeline, rows are sorted in case participants/sessions are out of order
-        pipelines_dict[label] = (
-            pipeline.sort_values(["participant_id", "session"])
-            .drop(["pipeline_name", "pipeline_version"], axis=1)
-            .reset_index(drop=True)
-        )
+    # TODO: Possibly temporary fix - to avoid related assessment columns from being out of order
+    sort = bool(schema == "imaging")
```

Comment:
I'm assuming sorting is done here now so it only happens for the "non pheno" use case - and because it was previously handled directly in the datatable? If I misunderstand and this is just unrelated and new functionality -> better to put in separate PR in future.

Reply:
Actually, the opposite! :) Previously, the non-subject-identifier columns in the datatable were automatically sorted alphabetically by the `groupby` command in this function, which was desirable when the columns were processing pipeline names. However, Shweta + Nikhil asked to _not_ sort the columns for phenotypic data since the order in which they appear in the input CSV is informative (i.e. rows (columns) corresponding to the subscales of the same assessment are adjacent).

The purpose of this addition is to enable the sort only when the bagel contains imaging data.